src/functions.cpp

```diff
@@ -1,10 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <algorithm>
 #include <iostream>
 #include <limits>
 
 void print(std::vector<int> const & numbers) {
-    for (int i = 0; i < numbers.size(); ++i) {
+    for (unsigned int i = 0; i < numbers.size(); ++i) {
```

Comment:
I recommend to use size_t here istead of unsigned int, that's why:
https://en.cppreference.com/w/cpp/types/size_t

Reply:
Thank you, I changed that